gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StatisticsTrackingHttpRequestInitializer.java

```diff
@@ -86,33 +53,12 @@ private void setHttpRequestFailureStats(HttpResponse response) {
       return;
     }
 
-    HttpRequest request = response.getRequest();
-    switch (request.getRequestMethod()) {
-      case "DELETE":
-        increment(HTTP_DELETE_REQUEST_FAILURE);
-        break;
-      case "GET":
-        increment(HTTP_GET_REQUEST_FAILURE);
-        break;
-      case "HEAD":
-        increment(HTTP_HEAD_REQUEST_FAILURE);
-        break;
-      case "PATCH":
-        increment(HTTP_PATCH_REQUEST_FAILURE);
-        break;
-      case "POST":
-        increment(HTTP_POST_REQUEST_FAILURE);
-        break;
-      case "PUT":
-        increment(HTTP_PUT_REQUEST_FAILURE);
-        break;
-      default:
-        throw new IllegalStateException("Unknown request method: " + request.getRequestMethod());
-    }
+    String requestMethod = Ascii.toUpperCase(response.getRequest().getRequestMethod());
```

Comment:
nit: consider refactoring this logic to reduce the code duplication.

Reply:
Didn't find elegant way to minimize code duplication, this seems like a sweet spot.